libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -0,0 +1,792 @@
+            return;
+        } else {
+            _last_send_us = now;
+        }
+        // scan for all ESCs in OneWire bus
+        _scan_active = scan_escs();
+        return;
+    }
+
+    if (_setup_active < MOTOR_COUNT_MAX) {
+        if (now - _last_send_us < DELAY_TIME_US) {
+            // setup should not be done too fast, as the bootloader has some message timing requirements.
+            return;
+        } else {
+            _last_send_us = now;
+        }
+        if (_found_escs_count == 0) {
+            _scan_active = 0;  // no ESC has been found yet, start scanning again
+        } else {
+            // check if in bootloader, start ESCs FW if they are and prepare fast-throttle command
+            _setup_active = init_escs();
+        }
+        return;
+    }
+
+    if (_set_full_telemetry_active < MOTOR_COUNT_MAX+1) {
+        if (now - _last_send_us < DELAY_TIME_US) {
+            // telemetry mode should not be done too fast, as the bootloader has some message timing requirements.
+            return;
+        } else {
+            _last_send_us = now;
+        }
+        // set telemetry to alternative mode (full telemetry from a single ESC, in a single response packet)
+        _set_full_telemetry_active = set_full_telemetry(1);
+        return;
+    }
+
+    _initialised = true;
+}
+
+/**
+  update configuration periodically to accommodate for parameter changes and test if the current configuration is OK
+*/
+void AP_FETtecOneWire::configuration_update()
+{
+    if (!hal.util->get_soft_armed()) { // configuration updates are only done when vehicle is disarmed
+
+        const uint32_t now = AP_HAL::millis();
+        if ((now - _last_config_update_ms > 2000) || _last_config_update_ms == 0) {  // only runs once every 2 seconds
+            _last_config_update_ms = now;
+
+            if (!_uart->is_dma_enabled()) {
+                GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "FTW UART needs DMA");
+                return; // will not work at all without DMA
+            }
+
+            // get the user-configured FETtec ESCs bitmask parameter
+            _mask = uint16_t(_motor_mask.get());
+            // tell SRV_Channels about ESC capabilities
+            SRV_Channels::set_digital_outputs(_mask, 0);
+
+            uint16_t smask = _mask;
+            _nr_escs_in_bitmask = 0;
+
+            // count the number of user-configured FETtec ESCs in the bitmask parameter
+            for (uint8_t i = 0; i < MOTOR_COUNT_MAX; i++) {
+                SRV_Channel* c = SRV_Channels::srv_channel(i);
+                if (c == nullptr || (smask & 0x01) == 0x00) {
+                    break;
+                }
+                smask >>= 1;
+                _nr_escs_in_bitmask++;
+            }
+
+#if HAL_WITH_ESC_TELEM
+            _update_rate_hz = AP::scheduler().get_loop_rate_hz();
+            _crc_error_rate_factor = 100.0f/(float)_update_rate_hz; //to save the division in loop, precalculate by the motor loops 100%/400Hz

```

Comment:
Here it is calculated per second, bellow it is used per 4 seconds...
I think there could be a problem with this.